server/src/routes.py

```diff
@@ -18,22 +22,73 @@
 def __replace_config_data(data: bytes) -> bytes:
     config = cfg()
 
+    # SIGNUP CONFIGURATION
     if not config.general.allow_signups:
         data = data.replace(b'name="PA-signup" content="true"', b'name="PA-signup" content="false"')
-    if not config.mail or not config.mail.enabled:
+    
+    # OIDC CONFIGURATION - Check first to determine if it's available
+    oidc_fully_configured = (
+        config.oidc.enabled and 
+        config.oidc.client_id and 
+        config.oidc.client_secret and
+        bool(config.oidc.domain or config.oidc.authorize_url)  # Either domain for discovery OR direct URLs
+    )
+    
+    # USERNAME/PASSWORD CONFIGURATION with fallback safety
+    # If username/password is disabled BUT OIDC is not available, enable username/password as fallback
+    username_pass_should_be_disabled = (
+        not config.general.username_password_enabled and 
+        oidc_fully_configured  # Only disable if OIDC is actually available
+    )
+    
+    if username_pass_should_be_disabled:
+        data = data.replace(b'name="PA-username-pass" content="true"', b'name="PA-username-pass" content="false"')
+    elif not config.general.username_password_enabled and not oidc_fully_configured:
+        # Fallback activated: username/password was disabled but OIDC is not available
+        # Keep username/password enabled to prevent users from being locked out
+        pass  # Meta tag stays as content="true"
+    
+    # MAIL CONFIGURATION
+    if config.mail is None or not config.mail.enabled or not config.mail.host or not config.mail.port or not config.mail.default_from_address:
```

Comment:
host, port and the default email cannot be empty if `config.mail` exists so checking them is unnecessary